SQL/proc_descriptions/test_assign_average_speed_to_all_segments_in_area.sql

```diff
@@ -0,0 +1,193 @@
+    , (3, ST_GeomFromText('POINT(5 5)', 4326), 1, false)
+    , (4, ST_GeomFromText('POINT(5 1)', 4326), 1, false)
+    , (5, ST_GeomFromText('POINT(9 1)', 4326), 1, false);
+    -- insert 3 nodes outside the test area
+    -- TODO: i've inserted 2 nodes outside test area and assigned to the test area, maybe we should add CHECK constraint with ST_GeomIntersects to the nodes table,
+    --  because later in this test, these nodes are included in calculations, which is not logical at all.
+    INSERT INTO nodes(id, geom, area, contracted) VALUES (6, ST_GeomFromText('POINT(11 1)', 4326), 1, false),
+    (7, ST_GeomFromText('POINT(11 5)', 4326), 1, false),
+    (8, ST_GeomFromText('POINT(11 11)', 4326), 1, false);
+
+    -- insert 2 ways within the test area
+    RAISE NOTICE 'Inserting into ways';
+    INSERT INTO ways(id, tags, geom, area, "from", "to", oneway) VALUES (1, '{example_tag => example}', ST_GeomFromText('LINESTRING(1 1, 1 5, 5 5)', 4326), 1, 1, 3, false),
+    (2, '{example_tag => example_2}', ST_GeomFromText('LINESTRING(5 5, 5 1, 9 1)', 4326), 1, 3, 5, false);
+
+    -- insert 1 way outside the test area
+    INSERT INTO ways(id, tags, geom, area, "from", "to", oneway) VALUES (3, '{example_tag => example_3}', ST_GeomFromText('LINESTRING(11 1, 11 5, 11 11)', 4326), 1, 6, 8, false);
+
+    -- insert 1 way with a segment outside the test area
+    INSERT INTO ways(id, tags, geom, area, "from", "to", oneway) VALUES (4, '{example_tag => example_4}', ST_GeomFromText('LINESTRING(1 1, 1 5, 11 5)', 4326), 1, 1, 7, false);
+
+    -- insert nodes_ways
+    RAISE NOTICE 'Inserting into nodes_ways';
+    INSERT INTO nodes_ways(way_id, node_id, position, area, id) VALUES -- create nodes_ways for every node in the way
+    (1, 1, 1, 1, 1),
+    (1, 2, 2, 1, 2),
+    (1, 3, 3, 1, 3),
+    (2, 3, 1, 1, 4),
+    (2, 4, 2, 1, 5),
+    (2, 5, 3, 1, 6),
+    (3, 6, 1, 1, 7),
+    (3, 7, 2, 1, 8),
+    (3, 8, 3, 1, 9),
+    (4, 1, 1, 1, 10),
+    (4, 2, 2, 1, 11),
+    (4, 7, 3, 1, 12);
+
+    -- create temporary table for expected output
+    CREATE TEMP TABLE expected_nodes_ways_speeds AS
+    SELECT * FROM nodes_ways_speeds WITH NO DATA;
+END;
+$$ LANGUAGE plpgsql;
+
+-- test case 1: testing both segments with matching speeds
+CREATE OR REPLACE FUNCTION setup_aastas_output_1() RETURNS VOID AS $$
+BEGIN
+    RAISE NOTICE 'Executing setup for test_aastas_output_1';
+    RAISE NOTICE 'Case: matching speeds';
+    -- insert 2 nodes_ways_speeds
+    -- TODO: add more records with different speeds
+    RAISE NOTICE 'Inserting into nodes_ways_speeds';
+    INSERT INTO nodes_ways_speeds(from_node_ways_id, speed, st_dev, to_node_ways_id, quality, source_records_count) VALUES
+                                                                                                                        (1, 70, 0, 1, 2, 1),
+                                                                                                                        (2, 75, 0, 2, 2, 1);
```

Comment:
I should've noticed it earlier, but did just now, when experimented with different values. It seems like procedure `assign_average_speed_to_all_segments_in_area` does not care what area nodes_ways_speeds records are related to, which seems highly illogical to me. The only thing it cares for is __speed__ and __st_dev__ from all records of `nodes_ways_speeds` table. 

In the end it just calculates average_speed from all nodes_ways_speeds, which could relate to __absolutely different areas__, and assigns it to nodes_ways, which __do__ relate to the given `area_id`. Im not sure that this is how it is supposed to work

Reply:
When assigning the average (fallback if speed for road segment is not present), the average is computed from all edges, irrespective of the area. That is the expected behavior